db/db.go

```diff
@@ -0,0 +1,129 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+// DB has all the tooling for talking to postgres
+type DB struct {
+	psql sq.StatementBuilderType
+}
+
+// New creates a DB instance
+func New(psql sq.StatementBuilderType) *DB {
+	return &DB{
+		psql: psql,
+	}
+}
+
+// Post represents a post in JSON
+type Post struct {
+	ID    int    `json:"id"`
+	Title string `json:"title"`
+	Body  string `json:"body"`
+}
+
+// GetPosts fetches all posts
+func (db *DB) GetPosts(ctx context.Context) (_ []Post, retErr error) {
+	posts := []Post{}
+
+	rows, err := db.psql.Select("*").From("posts").Query()
```

Comment:
I would encourage you to use `QueryContext` instead, which propagates the request context into the database layer. This gives you the ability to abort requests that have been started, and also will help when we want to do observability later.